pkg/api/types.go

```diff
@@ -379,7 +379,6 @@ type KubernetesConfig struct {
 	CloudControllerManagerConfig     map[string]string `json:"cloudControllerManagerConfig,omitempty"`
 	APIServerConfig                  map[string]string `json:"apiServerConfig,omitempty"`
 	SchedulerConfig                  map[string]string `json:"schedulerConfig,omitempty"`
-	PodSecurityPolicyConfig          map[string]string `json:"podSecurityPolicyConfig,omitempty"`
```

Comment:
To ensure back-compat, we can't remove these type definition properties. See `DockerEngineVersion` above: add a `// Deprecated` comment instead of actually deleting the property

Reply:
I see @jackfrancis! My bad :) I forget that this change would break the public API.
Even marking this as `Deprecated` - should we keep both implementations of receiving Raw PSP Manifests? Or just this new one (using addons raw definitions) ? Or just mark this property as deprecated and use the new approach

Reply:
We will need to eliminate the old implementation and then add back-compat tolerance for existing clusters. There is an `isUpgrade` flag on the defaults funcs in defaults.go. Basically, we want to check `if isUpgrade`, and if so, if `PodSecurityPolicyConfig` is not nil, then do the appropriate composition into the new addons implementation based on the data inside the "old" map[string]string

Reply:
Cool! Thanks for the detailed instructions 👍